### PR TITLE
YACL Config

### PR DESCRIPTION
**Pull request by AzureAaron**
Moves to YACL for the config library!

## Details
- Maintains the same config file format (nothing gets reset)
- We can move/relocate things in the UI without changing the file format :)
- YACL offers the ability to add custom functionality for the UI if we want

The Sonatype and Quilt repositories are also needed to retrieve YACL's dependencies

We also continue to use the old deprecated config api due to the new one being bugged as it can't deserialize maps.

## Changes
- Dungeons and Dwarven Mines are their own tab
  - This makes them work easier and their big enough to be their own tab (Dwarven Mines should be renamed to Mining tbh)
- The designs of some other tabs have been simplified a bit to work, minor change though
- The translation strings for quick nav buttons 2-12 are now unused except for the title ones (they aren't needed now)
- Buttons to access separate config screens have been added 

## Translation String Format
Since we're free to define the format of translations strings they should be updated after this PR (avoids major merge conflicts)

Additionally, some strings are now unused but have not yet been removed, the removal can happen with the string format change.

### CODE REVIEW

File: src/main/java/me/xmrvizzy/skyblocker/skyblock/shortcut/ShortcutsConfigScreen.java [hunk 1]
```diff
@@ -89,13 +95,15 @@ public void close() {
         if (client != null && shortcutsConfigListWidget.hasChanges()) {
             client.setScreen(new ConfirmScreen(confirmedAction -> {
                 if (confirmedAction) {
-                    super.close();
+                	this.client.setScreen(parent);
+                    if (parent == null) super.close();
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/skyblock/shortcut/ShortcutsConfigScreen.java:99]:
```suggestion
```

File: src/main/java/me/xmrvizzy/skyblocker/skyblock/shortcut/ShortcutsConfigScreen.java [hunk 2]
```diff
@@ -89,13 +95,15 @@ public void close() {
...
                 } else {
                     client.setScreen(this);
                 }
-            }, Text.translatable("text.cloth-config.quit_config"), Text.translatable("text.cloth-config.quit_config_sure"), Text.translatable("text.cloth-config.quit_discard"), ScreenTexts.CANCEL));
+            }, Text.translatable("text.skyblocker.quit_config"), Text.translatable("text.skyblocker.quit_config_sure"), Text.translatable("text.skyblocker.quit_discard"), ScreenTexts.CANCEL));
         } else {
-            super.close();
+            this.client.setScreen(parent);
+            if (parent == null) super.close();
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/skyblock/shortcut/ShortcutsConfigScreen.java:106]:
```suggestion
```

**Review by kevinthegreat1 — COMMENTED** [review]
Took a look at the code. Didn't confirm the config screens, categories, and translations.

File: src/main/java/me/xmrvizzy/skyblocker/skyblock/dungeon/DungeonMapConfigScreen.java [hunk 3]
```diff
@@ -51,7 +57,14 @@ public boolean mouseClicked(double mouseX, double mouseY, int button) {
...
-		AutoConfig.getConfigHolder(SkyblockerConfig.class).save();
+		SkyblockerConfig.save();
+		
+		if (parent != null) {
+			this.client.setScreen(parent);
+			
+			return;
+		}
+		
 		super.close();
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/skyblock/dungeon/DungeonMapConfigScreen.java:68]:
```suggestion
		this.client.setScreen(parent);
```

File: src/main/java/me/xmrvizzy/skyblocker/utils/render/title/TitleContainerConfigScreen.java [hunk 4]
```diff
@@ -158,7 +159,7 @@ public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
     public void close() {
         SkyblockerConfig.get().general.titleContainer.x = (int) hudX;
         SkyblockerConfig.get().general.titleContainer.y = (int) hudY;
-        AutoConfig.getConfigHolder(SkyblockerConfig.class).save();
+        SkyblockerConfig.save();
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/utils/render/title/TitleContainerConfigScreen.java:162]: Do we need to make this go back to the parent screen too?

Comment (AzureAaron) [reply src/main/java/me/xmrvizzy/skyblocker/utils/render/title/TitleContainerConfigScreen.java:162]: I forgot about this one, I'll add the button for it and that functionality

File: src/main/java/me/xmrvizzy/skyblocker/config/SkyblockerConfigManager.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+	}
+	
+	public static void save() {
+		INSTANCE.save();
+	}
+	
+	public static Screen createGUI(Screen parent) {
+		return YetAnotherConfigLib.create(INSTANCE, (defaults, config, builder) -> {
+			return builder
+					.title(Text.literal("Skyblocker"))
```
Comment (Grayray75) [inline src/main/java/me/xmrvizzy/skyblocker/config/SkyblockerConfigManager.java:65]:
```suggestion
					.title(Text.translatable("text.autoconfig.skyblocker.title"))
```

this should be translatable

**Review by kevinthegreat1 — APPROVED** [review]
Looked through the config screen and it looks good.

**Review by Grayray75 — APPROVED** [review]

**Review by kevinthegreat1 — DISMISSED** [review]
LGTM.

**Review by kevinthegreat1 — APPROVED** [review]

**Pull request merged**